### PR TITLE
VxWorks libc cleanups and fixes for some libc-test issues that were p…

### PR DESCRIPTION
**Pull request by n-salim**
…reviously  skipped

### CODE REVIEW

File: src/vxworks/mod.rs [hunk 1]
```diff
@@ -1340,13 +1310,25 @@ extern "C" {
...
         handle: *mut ::c_void,
         symbol: *const ::c_char,
     ) -> *mut ::c_void;
+
+    #[link_name = "_rtld_dlclose"]
     pub fn dlclose(handle: *mut ::c_void) -> ::c_int;
+
+    #[link_name = "_rtld_dladdr"]
+    pub fn dladdr(addr: *mut ::c_void, info: *mut Dl_info) -> ::c_int;
+
```
Comment (BaoshanPang) [inline src/vxworks/mod.rs]: src/vxworks/mod.rs:1331 - trailing whitespace

I guess you didn't run "sh ci/style.sh".

**Review by BaoshanPang — CHANGES_REQUESTED** [review]
Please run "sh ci/style.sh" to find the style and format problems.

**Comment by BaoshanPang** [comment]
I am seeing these errors in building rust with this branch:

error[E0412]: cannot find type `passwd` in crate `libc`
   --> src/libstd/sys/vxworks/os.rs:300:31
    |
300 |         let mut passwd: libc::passwd = mem::zeroed();
    |                               ^^^^^^ not found in `libc`

error[E0425]: cannot find function `getpwuid_r` in crate `libc`
   --> src/libstd/sys/vxworks/os.rs:302:21
    |
302 |         match libc::getpwuid_r(libc::getuid(), &mut passwd, buf.as_mut_ptr(),
    |                     ^^^^^^^^^^ not found in `libc`

**Comment by BaoshanPang** [comment]
> I am seeing these errors in building rust with this branch:
> 
> error[E0412]: cannot find type `passwd` in crate `libc`
> --> src/libstd/sys/vxworks/os.rs:300:31
> |
> 300 | let mut passwd: libc::passwd = mem::zeroed();
> | ^^^^^^ not found in `libc`
> 
> error[E0425]: cannot find function `getpwuid_r` in crate `libc`
> --> src/libstd/sys/vxworks/os.rs:302:21
> |
> 302 | match libc::getpwuid_r(libc::getuid(), &mut passwd, buf.as_mut_ptr(),
> | ^^^^^^^^^^ not found in `libc`

These functions are used in function home_dir. Do we need to support this function?  It seems there is no concept of "home directory" on vxWorks.

File: src/vxworks/mod.rs (hunk 1 above)
Comment (n-salim) [reply src/vxworks/mod.rs]: Sorry I'm setting a very bad example with this untested change :( Thanks for your diligence.

Ran ci/style.sh and ci/build.sh

Comment (n-salim) [reply src/vxworks/mod.rs]: ci/build.sh is still running so please wait until I report back ....

**Comment by n-salim** [comment]
> These functions are used in home_dir. Do we need to support this function? It seems there is no concept of "home directory" on vxWorks.

Resolved by https://github.com/Wind-River/rust/pull/24

File: src/vxworks/mod.rs (hunk 1 above)
Comment (n-salim) [reply src/vxworks/mod.rs]: OK it finished and passed. Pity we can't just run it for the VxWorks targets that come right at the end of the list, the rest are a waste of time unless we change some of the common code!

Comment (BaoshanPang) [reply src/vxworks/mod.rs]: I have simplified version that only run vxworks targets:
/folk/prj-rust-dev/bpang/wr-rust/scripts/test/vxbuild.sh

And I am thinking about to add a "FILTER" to script build.sh so we can choose which target(s) to run.

**Comment by BaoshanPang** [comment]
Can you rebase/squash the two commits into one? This is usually what I do.

**Comment by n-salim** [comment]
Hold on, something went wrong and the style test seems to be reporting problems. Let me figure it out ...

**Review by BaoshanPang — APPROVED** [review]

**Comment by n-salim** [comment]
OK ci/style.sh passes now. Actually it may be that I was running a newer version of rustfmt ths time.

**Pull request merged**